hermeto/core/package_managers/pip.py

```diff
@@ -1799,14 +1796,15 @@ def _validate_requirements(requirements: list[PipRequirement]) -> None:
 
         # Fail if VCS requirement uses any VCS other than git or does not have a valid ref
         elif req.kind == "vcs":
-            url = urllib.parse.urlparse(req.url)
+            url = parse.urlparse(req.url)
 
             if not url.scheme.startswith("git"):
                 raise UnsupportedFeature(
                     f"Unsupported VCS for {req.download_line}: {url.scheme} (only git is supported)"
                 )
 
-            if not GIT_REF_IN_PATH.search(url.path):
+            git_ref_in_path = re.compile(r"@[a-fA-F0-9]{40}$")
```

Comment:
This particular change wasn't requested though. What this does is it'll compile the regex over and over again for every VCS `req` in the requirements list instead of just applying a pre-compiled one. It's definitely true that we don't expect many VCS type of requirements, so probably not a big loss if it's used **iff** a VCS req exists, but:
- you're already stretching it with the unrelated changes stuffed into this one particular commit to the point where a standalone commit should be introduced :)
- if matching a regex is a one-off task the call to `compile` is pointless because all `search`-like `re` APIs will compile the pattern themselves, IOW either do it once on the global level like it were before or just issue `re.search` directly